### PR TITLE
[Core] Add GetValueOnIntegrationPoints to Python

### PR DESCRIPTION
**Pull request by tteschemacher**
Hi all,

I am entering the world of beauty. I know that this was already a long and bad discussion, but I need it. It might confuse with the GetValue**S**OnIntegrationPoints which is wrongly calling the CalculateOnIntegrationPoints. 

This calls the correct function and comes without the controversial additional s.

Feel free to blame me. But I need this to make the MPM run again.

Cheers!

### CODE REVIEW

**Review by loumalouomega — APPROVED** [review]
This approval should be granted

**Review by RiccardoRossi — CHANGES_REQUESTED** [review]
i am blocking since i don't understand what is going on here.

if we add GetValuesOnIntegrationPoints we need to remove the version without the s...

**Comment by RiccardoRossi** [comment]
well, the function with the s (which should be called CalculateValueOnIntegrationPoints) is only used in 3 files, so we could just rename it here

~~~
grep -r GetValuesOnIntegrationPoints --include=*.py
applications/PfemSolidMechanicsApplication/python_scripts/gauss_point_utility.py:            cauchy_vector = elem.GetValuesOnIntegrationPoints( KratosMultiphysics.CAUCHY_STRESS_TENSOR, SomeProcessInfo)
applications/PfemSolidMechanicsApplication/python_scripts/gauss_point_utility.py:            epsi_vector = elem.GetValuesOnIntegrationPoints( KratosMultiphysics.GREEN_LAGRANGE_STRAIN_TENSOR, SomeProcessInfo)
applications/FemToDemApplication/python_scripts/MainCouplingFemDem.py:                    stress_tensor = Elem.GetValuesOnIntegrationPoints(KratosFemDem.STRESS_VECTOR_INTEGRATED, self.FEM_Solution.main_model_part.ProcessInfo)
~~~

File: kratos/python/add_mesh_to_python.cpp [hunk 1]
```diff
@@ -593,11 +700,19 @@ void  AddMeshToPython(pybind11::module& m)
     .def("CalculateOnIntegrationPoints", CalculateOnIntegrationPointsArray1d)
     .def("CalculateOnIntegrationPoints", CalculateOnIntegrationPointsVector)
     .def("CalculateOnIntegrationPoints", CalculateOnIntegrationPointsMatrix)
+    // GetValue  s  OnIntegrationPoints
```
Comment (RiccardoRossi) [inline kratos/python/add_mesh_to_python.cpp:703]: my proposal would be to rename the GetValueS... to CalculateValueS... and to also change it in the 3files that make use of it. 

Other than that i think we simply need to keep track of the change in 
https://github.com/KratosMultiphysics/Kratos/projects/26

and to mark the current PR as APIBreaker

File: kratos/python/add_mesh_to_python.cpp [hunk 2]
```diff
@@ -374,6 +374,113 @@ pybind11::list GetValuesOnIntegrationPointsMatrix( TObject& dummy,
...
+template< class TObject >
+pybind11::list GetValueOnIntegrationPointsMatrix(
+    TObject& dummy,
+    const Variable<Matrix>& rVariable,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    pybind11::list values_list;
+    size_t nb_integration_points = dummy.GetGeometry().IntegrationPointsNumber();
+    std::vector<Matrix> values(nb_integration_points);
+    dummy.GetValueOnIntegrationPoints(rVariable, values, rCurrentProcessInfo);
```
Comment (philbucher) [inline kratos/python/add_mesh_to_python.cpp:468]: this should be mapped to `CalculateOnIntegrationPoints` I would say, its what we wanted to do in #2194

**Review by philbucher — CHANGES_REQUESTED** [review]
if we go ahead with this one then I would say it should be as proposed in #2194 
=> `CalculateOnIntegrationPoints`

All elements in Structural implement both, `CalculateOnIntegrationPoints` and `GetValuesIntegrationPoints`, where the later is only a wrapper to call `CalculateOnIntegrationPoints`

**Comment by tteschemacher** [comment]
Guys. That are NO productive comments.

This is related to a hot fix to make the ParticleMechanics run again.

**Comment by philbucher** [comment]
> Guys. That are NO productive comments.
> 
> This is related to a hot fix to make the ParticleMechanics run again.

I don't agree, I gave constructive criticism
This is big changes in the core after all!

In any case, @RiccardoRossi will post sth in the next minutes

**Comment by RiccardoRossi** [comment]
@tteschemacher the point is that what you propose is NOT a hotfix.
you are adding an interface that will remain there forever, without fixing the underlying issue...

now, we definitely do not want u to make an enormous change through the Kratos to fix the mpm.

my proposal is the following:
1 - in the MPM use CalculateOnIntergrationPoint (i don't know if there is the "s" or not, i am referring to the function defined in the element API). My point is to use the function called Calculate instead of the one called Get
2 - Let's expose to python ONLY the function CalculateOnIntegrationPoints, and remove the other functions. This way you will also not need to add your new interface, but just to change the name to a few functions
3 - no function "Get*OnIntegrationPoint" is exposed at all...but you don't need since you will use CalculateOnIntegrationPoints instead.

as far as i can see 2 only requires changing 3 python files:

~~~
grep -r GetValuesOnIntegrationPoints --include=*.py
applications/PfemSolidMechanicsApplication/python_scripts/gauss_point_utility.py:            cauchy_vector = elem.GetValuesOnIntegrationPoints( KratosMultiphysics.CAUCHY_STRESS_TENSOR, SomeProcessInfo)
applications/PfemSolidMechanicsApplication/python_scripts/gauss_point_utility.py:            epsi_vector = elem.GetValuesOnIntegrationPoints( KratosMultiphysics.GREEN_LAGRANGE_STRAIN_TENSOR, SomeProcessInfo)
applications/FemToDemApplication/python_scripts/MainCouplingFemDem.py:   
~~~

@tteschemacher  does this make sense to you?

**Comment by tteschemacher** [comment]
I am closing this PR as it was exchanged by #6528 .

**Pull request closed without merging**